### PR TITLE
Add initial cache node registration

### PR DESCRIPTION
**Pull request by bryanlb**
Adds initial node type, run config, associated configs, and README updates.

Split from https://github.com/slackhq/kaldb/pull/117

### CODE REVIEW

File: kaldb/src/main/java/com/slack/kaldb/metadata/cache/CacheSlotMetadata.java [hunk 1]
```diff
@@ -4,6 +4,7 @@
 import com.slack.kaldb.proto.metadata.Metadata;
 
 public class CacheSlotMetadata extends KaldbMetadata {
+  public static final String METADATA_SLOT_NAME = "SLOT";
```
Comment (mansu) [inline kaldb/src/main/java/com/slack/kaldb/metadata/cache/CacheSlotMetadata.java:7]: Seems like an unused variable?

File: README.md [hunk 2]
```diff
@@ -1,13 +1,26 @@
...
 # Development
 
 To build the binary: `mvn clean package`
 
-IntelliJ: Import the project as a maven project.
+## Local development
+
+> IntelliJ: Import the project as a Maven project.
+
+IntelliJ run configs are provided for all node types, and execute using the provided `config/config.yaml`. To start the
```
Comment (mansu) [inline README.md:21]: Please also mention that the run configs are in .run folder.

File: .run/KalDb_cache.run.xml [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+      <env name="NODE_ROLES" value="CACHE" />
+    </envs>
+    <option name="MAIN_CLASS_NAME" value="com.slack.kaldb.server.Kaldb" />
+    <module name="kaldb" />
+    <option name="PROGRAM_PARAMETERS" value="config/config.yaml" />
+    <method v="2">
+      <option name="Make" enabled="true" />
+    </method>
+  </configuration>
+</component>
```
Comment (mansu) [inline .run/KalDb_cache.run.xml:13]: new line?

**Review by mansu — CHANGES_REQUESTED** [review]

File: .run/KalDb_cache.run.xml (hunk 3 above)
Comment (bryanlb) [reply .run/KalDb_cache.run.xml:13]: This is an auto-generated file, and is synced with IntelliJ changes so we shouldn't edit the contents.

File: kaldb/src/main/java/com/slack/kaldb/metadata/cache/CacheSlotMetadata.java (hunk 1 above)
Comment (bryanlb) [reply kaldb/src/main/java/com/slack/kaldb/metadata/cache/CacheSlotMetadata.java:7]: Yeah, it's just due to how I split this off of the previous PR. Both of the cache slot string variables will be used in the `ChunkManager` refactor. I put them into this PR so the `ChunkManager` refactor has to touch fewer classes, as that's gonna be a pretty big update.

File: README.md (hunk 2 above)
Comment (bryanlb) [reply README.md]: _Done_

**Review by vthacker — APPROVED** [review]
+1 

I haven't looked at it super closely ( juggling a few other things ) But if I have doubts I'll check with you in the next few days

**Review by mansu — APPROVED** [review]

**Pull request merged**